e2e/LongHaul/service/IotHub.cs

```diff
@@ -96,16 +96,19 @@ await foreach (ClientTwin device in allDevices)
 
                         async Task Operations()
                         {
-                            var deviceOperations = new DeviceOperations(s_serviceClient, deviceId, _logger.Clone());
+                            var deviceOperations = new DeviceOperations(s_serviceClient, deviceId);
                             _logger.Trace($"Creating {nameof(DeviceOperations)} on the device [{deviceId}]", TraceSeverity.Verbose);
 
+                            Logger loggerPerDevice = _logger.Clone();
+                            loggerPerDevice.LoggerContext.Add("DeviceId", deviceId);
```

Comment:
Create new logger clone per device, add custom dimension of "DeviceId", pass its clone to each operation.

Reply:
We're using camelCase for properties, right? Also, this should be put in a logging constants file.